### ISSUE TITLE
When used together with FlatList as per instruction, RefreshControl renders under the navigation header.

### ISSUE THREAD

**Issue by joonhocho** [benevbright/react-navigation-collapsible#138]
## Information
- react-native version: 0.62.2 (latest)
- react-navigation version: 5.5.0 (latest)
- react-navigation-collapsible version: 5.6.1 (latest)
- Platform (iOS/Android): iOS
- react-native init or Expo: react-native init

## Detail
When used together with FlatList as per instruction, RefreshControl renders under the navigation header.
<img width="709" alt="Screen Shot 2020-06-02 at 2 52 57 AM" src="https://user-images.githubusercontent.com/930795/83506652-4d6b5080-a47c-11ea-9053-3ebddfdbb5f7.png">

**Comment by amin79** [comment]
@joonhocho I have same problem. Did you find any solution?

**Comment by benevbright** [comment]
There is `progressViewOffset` for `FlatList` but it's only for Android. 
It seems to be a very old issue of RN. https://react-native.canny.io/feature-requests/p/offsetting-refreshcontrol-on-ios-via-progressviewoffset

I will get back to here if I find a workaround or a solution.

**Comment by benevbright** [comment]
https://stackoverflow.com/questions/56969502/progressviewoffset-for-ios-refresh-control-react-native This SOF answer says that we can use `contentInset`.

**Comment by isilher** [comment]
I encountered the same problem. Replacing the padding with `contentInset` *only* for iOS and creating a custom RefreshControl with `progressViewOffset` *only* for android worked on a standard FlatList. 

However, I then ran into problems with my SectionList: the section header was also rendered with the contentInset. I did not manage to counter this and was not satisfied with all the platform specific exceptions in my components.

As an alternative solution, I applied the `translateY` animated value from the `useCollapsibleStack` hook to my SectionList as a whole. This in turn required me to apply the `containerPaddingTop` to the contentContainer *bottom* padding to prevent the list from getting cut off at the bottom.

``` tsx
  const {
    onScroll,
    scrollIndicatorInsetTop,
    containerPaddingTop,
    translateY,
  } = useCollapsibleStack()

  return (
    <Animated.SectionList
      // other props
      onScroll={onScroll}
      scrollIndicatorInsets={{ bottom: scrollIndicatorInsetTop }}
      contentContainerStyle={{ paddingBottom: containerPaddingTop }}
      scrollEventThrottle={16} // without this, it can get jittery.
      style={{
        transform: [
          {
            translateY: translateY.interpolate({
              inputRange: [0, containerPaddingTop],
              outputRange: [containerPaddingTop, 2 * containerPaddingTop],
            }),
          },
        ],
      }}
    />
  )
```

**Comment by saadi-ninjasCode** [comment]
> I encountered the same problem. Replacing the padding with `contentInset` _only_ for iOS and creating a custom RefreshControl with `progressViewOffset` _only_ for android worked on a standard FlatList.
> 
> However, I then ran into problems with my SectionList: the section header was also rendered with the contentInset. I did not manage to counter this and was not satisfied with all the platform specific exceptions in my components.
> 
> As an alternative solution, I applied the `translateY` animated value from the `useCollapsibleStack` hook to my SectionList as a whole. This in turn required me to apply the `containerPaddingTop` to the contentContainer _bottom_ padding to prevent the list from getting cut off at the bottom.
> 
> ```tsx
>   const {
>     onScroll,
>     scrollIndicatorInsetTop,
>     containerPaddingTop,
>     translateY,
>   } = useCollapsibleStack()
> 
>   return (
>     <Animated.SectionList
>       // other props
>       onScroll={onScroll}
>       scrollIndicatorInsets={{ bottom: scrollIndicatorInsetTop }}
>       contentContainerStyle={{ paddingBottom: containerPaddingTop }}
>       scrollEventThrottle={16} // without this, it can get jittery.
>       style={{
>         transform: [
>           {
>             translateY: translateY.interpolate({
>               inputRange: [0, containerPaddingTop],
>               outputRange: [containerPaddingTop, 2 * containerPaddingTop],
>             }),
>           },
>         ],
>       }}
>     />
>   )
> ```

Android: It starts fluctuating / jitter screen while scrolling even different values of scrollEventtrottle is not solving it.

**Comment by isilher** [comment]
You're right! I'm not sure how I missed this on my initial tests but it indeed jitters quite a lot. It seems the android scroll gesture does not like us moving the list up and down during the scroll motion.

I did not find a good common solution so I ended up splitting my component into:

**AnimatedSectionList.ios.tsx** which uses the above transelateY hack.

**AnimatedSectionList.android.tsx** which uses the default collapsible header config and adds the `progressViewOffset` to a custom `refreshControl` as suggested by Bright.

**Comment by benevbright** [comment]
Sorry about it. 
FYI, the jittering issue is RN's very old bug. https://github.com/facebook/react-native/issues/21801
If this bug was not present, this library would have had a lot more potential because it could have built in a much better way than using `absolute` position header. I only can tell you there are many limitations in this library.

**Comment by wayne-tt** [comment]
if you use FlatList, use progressViewOffset prop.
